lib/lbt/analyzer/SmartTemplateAnalyzer.js

```diff
@@ -134,24 +134,39 @@ class TemplateComponentAnalyzer {
 			const TA = defineCall.findImportName("sap/suite/ui/generic/template/lib/TemplateAssembler.js");
 			// console.log("local name for TemplateAssembler: %s", TA);
 			if ( TA && defineCall.factory ) {
-				defineCall.factory.body.body.forEach( (stmt) => {
-					if ( stmt.type === Syntax.ReturnStatement &&
-							isMethodCall(stmt.argument, [TA, "getTemplateComponent"]) &&
-							stmt.argument.arguments.length > 2 &&
-							stmt.argument.arguments[2].type === "ObjectExpression" ) {
-						templateName = this._analyzeTemplateClassDefinition(stmt.argument.arguments[2]) || templateName;
+				if (defineCall.factory.type === Syntax.ArrowFunctionExpression &&
+					defineCall.factory.expression === true) {
+					if ( this._isTemplateClassDefinition(TA, defineCall.factory.body) ) {
+						templateName =
+							this._analyzeTemplateClassDefinition(defineCall.factory.body.arguments[2]) || templateName;
 					}
-				});
+				} else {
+					defineCall.factory.body.body.forEach( (stmt) => {
+						if ( stmt.type === Syntax.ReturnStatement &&
+							this._isTemplateClassDefinition(TA, stmt.argument)
+						) {
+							templateName =
+								this._analyzeTemplateClassDefinition(stmt.argument.arguments[2]) || templateName;
+						}
+					});
+				}
+				if (defineCall.factory.async) {
```

Comment:
This message is IMO misleading. The ui5loader can handle async callbacks. The returned promise is just used as export value and delivered to anyone importing the module. 

Here, the smart templates (FE v2) cannot handle such a promise. That's the reason why we should log a message.

Reply:
Let's handle this in a follow-up PR, so that we can merge the current changes.